drivers/gpio/gpio_sx1509b.c

```diff
@@ -254,16 +254,28 @@ static int port_get(struct device *dev,
 
 static int port_write(struct device *dev,
 		      gpio_port_pins_t mask,
-		      gpio_port_value_t value)
+		      gpio_port_value_t value,
+		      gpio_port_value_t toggle)
 {
+	/* Can't do I2C bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
 	const struct sx1509b_config *cfg = dev->config->config_info;
 	struct sx1509b_drv_data *drv_data = dev->driver_data;
 	u16_t *outp = &drv_data->pin_state.data;
 
 	k_sem_take(&drv_data->lock, K_FOREVER);
 
 	u16_t orig_out = *outp;
-	u16_t out = (orig_out & ~mask) | (u16_t)(value & mask);
+	u16_t out;
+
+	if (toggle) {
+		out = orig_out ^ toggle;
+	} else {
+		out = (orig_out & ~mask) | (value & mask);
+	}
```

Comment:
This is a local function so the API doesn't need to be perfect. However, it may be cleaner if we wrote this bit as
```
	out = (orig_out & ~mask) | (value & mask);
	out = out ^ toggle;
```
This would be more in line with what the function parameter list suggests. Only the call in `port_toggle_bits` would need to be changed to `port_write(dev, 0, 0, pins);`. Just a suggestion.

Reply:
Thanks, I'd missed that setting `mask` to zero makes it work the natural way.